main_test.go

```diff
@@ -507,6 +507,71 @@ func TestGenerate(t *testing.T) {
       "zone": "uscentral-5"
     }
   }
+}`,
+		},
+		{
+			desc: "happy case with federation support in directpath with new xdstp style name",
+			input: configInput{
+				xdsServerUri:               "trafficdirector.googleapis.com:443",
+				gcpProjectNumber:           123456789012345,
+				vpcNetworkName:             "thedefault",
+				ip:                         "10.9.8.7",
+				zone:                       "uscentral-5",
+				includeV3Features:          true,
+				includeFederationSupport:   true,
+				includeDirectPathAuthority: true,
+				ipv6Capable:                true,
+				includeXDSTPNameInLDS:      true,
+			},
+			wantOutput: `{
+  "xds_servers": [
+    {
+      "server_uri": "trafficdirector.googleapis.com:443",
+      "channel_creds": [
+        {
+          "type": "google_default"
+        }
+      ],
+      "server_features": [
+        "xds_v3"
+      ]
+    }
+  ],
+  "authorities": {
+    "": {},
+    "traffic-director-c2p.xds.googleapis.com": {
+      "xds_servers": [
+        {
+          "server_uri": "dns:///directpath-pa.googleapis.com",
+          "channel_creds": [
+            {
+              "type": "google_default"
+            }
+          ],
+          "server_features": [
+            "xds_v3"
+          ]
+        }
+      ],
+      "client_listener_resource_name_template": "xdstp://traffic-director-c2p.xds.googleapis.com/envoy.config.listener.v3.Listener/%s"
+    },
+    "trafficdirector.googleapis.com:443": {
```

Comment:
From my "Oh boy" comment, it seems that this needs to be traffic-director-global.xds.googleapis.com.

Reply:
I've got it updated now. Sorry for the back and forth.